rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -104,15 +106,15 @@ local function is_backend_with_external_name(backend)
 end
 
 local function sync_backend(backend)
+  if is_backend_with_external_name(backend) then
+    backend = resolve_external_names(backend)
+  end
```

Comment:
It would be good to leave a comment on why this comes before line 113-116 so that it's clear in future work that this must come here.